components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/util/CommonUtils.java

```diff
@@ -18,10 +18,56 @@
 
 package org.wso2.bfsi.consent.management.common.util;
 
+import com.nimbusds.jose.JWSObject;
+import net.minidev.json.JSONObject;
+import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.RequestObjectException;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
 
 /**
  * Common utility methods.
  */
 public class CommonUtils {
 
+    /**
+     * Decode request JWT.
+     *
+     * @param jwtToken jwt sent by the tpp
+     * @param jwtPart  expected jwt part (header, body)
+     * @return json object containing requested jwt part
+     * @throws java.text.ParseException if an error occurs while parsing the jwt
+     */
+    public static JSONObject decodeRequestJWT(String jwtToken, String jwtPart) throws java.text.ParseException {
+
+        JWSObject plainObject = JWSObject.parse(jwtToken);
+
+        if ("head".equals(jwtPart)) {
+            return plainObject.getHeader().toJSONObject();
+        } else if ("body".equals(jwtPart)) {
+            return plainObject.getPayload().toJSONObject();
+        }
+        return null;
+    }
+
+    /**
+     * Check whether the client ID belongs to a regulatory app.
+     * @param clientId  client ID
+     * @return true if the client ID belongs to a regulatory app
+     * @throws RequestObjectException If an error occurs while checking the client ID
+     */
+    @Generated(message = "Excluding from code coverage since it requires a service call")
+    public static boolean isRegulatoryApp(String clientId) throws RequestObjectException {
+
+        try {
+            return OAuth2Util.isFapiConformantApp(clientId);
```

Comment:
why are we using FAPI to check if its a regulatory app or not? Does this mean all FAPI compliant apps are regulatory?

Reply:
Yes, Fapi complaint apps are the regulatory apps in IS 7.0